Outline.md

```diff
@@ -0,0 +1,137 @@
+# Outline - Build Modern APIs in NodeJS
+
+Welcome to the course, in this course we are going to learn how to build modern APIs using NodeJS. Mainly we will focus on REST API as it will be found all over the world. But we will also try to understand other popular competitors of REST APIs like GraphQL, JSON API, gRPC like techniques. End of this course you will learn - 
+
+-   What is API (Application Programming Interface)
+-   What is REST API (Representation State Transfer)
+-   Why do We Need API
+-   Best Alternatives of REST API and Their Use Cases
+-   How to Implement REST API using Node and Express
+-   How to Implement GraphQL API using GraphQL Yoga or Express GraphQL
+-   JWT Authentication, Authorization & API Security
+-   Design Principles or Best Practices of API Development
+-   API Version Control and Documentation
+
+
+
+This course will be the foundation of your backend journey using **NodeJS** and **ExpressJS** two hot most and demandable technologies. If you never build any server side application yet don't worry, you are going to love the backend development process. NodeJS makes things easy for us. After completing this course you will never rely on AWS Amplify, Firebase or Superbase. You can build your own backend for your future frontend, mobile application or Saas projects.
+
+
+
+## Prerequisites 
+
+To get 100% benefited from this course you must have some previous skills on programming. You need -
+
+-   Working knowledge of JavaScript and ES6+ (ESNext)
+-   Asynchronous Programming in JavaScript
+-   JavaScript Promise and Async Await
+-   Basic Understanding of Web
+
+
+
+## Lessons
+
+I will update describe every chapter later after your confirmation. 
+
+
+
+1.  Introduction
+    1.  Course Introduction (Slide)
```

Comment:
We should be sure to add a "what this course is NOT" slide / video to indicate up front that this is not going to be a full stack / database course.

Reply:
Sure, I will add that